core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js

```diff
@@ -0,0 +1,468 @@
+                  thirdCollectionNode]);
+      expect(CollectionLinearizerService.removeCollectionNode(collection,
+          'exp_id0')).toBe(true);
+      expect(collection.containsCollectionNode('exp_id0')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([secondCollectionNode, thirdCollectionNode]);
+    });
+
+    it('should correctly remove the last node from a collection', function() {
+      var collection = createLinearCollection();
+      expect(collection.containsCollectionNode('exp_id2')).toBe(true);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+      expect(CollectionLinearizerService.removeCollectionNode(collection,
+          'exp_id2')).toBe(true);
+      expect(collection.containsCollectionNode('exp_id2')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode]);
+    });
+
+    it('should correctly remove a middle node from a collection', function() {
+      var collection = createLinearCollection();
+      expect(collection.containsCollectionNode('exp_id1')).toBe(true);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+      expect(CollectionLinearizerService.removeCollectionNode(collection,
+          'exp_id1')).toBe(true);
+      expect(collection.containsCollectionNode('exp_id1')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, thirdCollectionNode]);
+    });
+  });
+
+  describe('appendCollectionNode()', function() {
+    it('should correctly append a node to an empty collection', function() {
+      var collection = CollectionObjectFactory.createEmptyCollection();
+      expect(collection.containsCollectionNode('exp_id0')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+                  collection)).toEqual([]);
+      expect(firstCollectionNode.getAcquiredSkillList().getSkills()).toEqual(
+          []);
+      CollectionLinearizerService.appendCollectionNode(collection, 'exp_id0',
+          firstCollectionNode.getExplorationSummaryObject());
+      firstCollectionNode = collection.getCollectionNodeByExplorationId(
+          'exp_id0');
+      expect(
+          firstCollectionNode.getPrerequisiteSkillList().getSkills()).toEqual(
+              []);
+      expect(firstCollectionNode.getAcquiredSkillList().getSkills()).toEqual([
+          'exp title0']);
```

Comment:
I don't think it needs to be. What's important is that the linearizer service perform the correct results when given a certain input. If, in the future, we decide to change how we use the skill lists when performing this functionality then we wouldn't want to break these more fragile tests. The linearizer service could still be operating correctly, even though its implementation changed.